### PR TITLE
Own Cluster Update Correction

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

When sending and update with `own_cluster` plan the update failed when `sm_credentials` were provided in the request. The following pull requests fixes that bug.

Change in the PR:
* update bug fix,
* corrected tests to include json marshalling,
* corrected fake provisioner client to return only instance previously persisted in the client.

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/cmd/broker/broker_suite_test.go [hunk 1]
```diff
@@ -515,7 +515,7 @@ func (s *BrokerSuiteTest) FinishProvisioningOperationByReconciler(operationID st
 		if err != nil {
 			return false, nil
 		}
-		if op.ProvisionerOperationID != "" {
+		if op.ProvisionerOperationID != "" || op.ProvisioningParameters.PlanID == broker.OwnClusterPlanID {
```
Comment (jaroslaw-pieszka) [inline components/kyma-environment-broker/cmd/broker/broker_suite_test.go:518]: There is helper function `isOwnClusterPlan`

File: components/kyma-environment-broker/internal/storage/driver/memory/instance.go [hunk 2]
```diff
@@ -153,15 +154,24 @@ func (s *instances) GetByID(instanceID string) (*internal.Instance, error) {
 		return nil, dberr.NotFound("instance with id %s not exist", instanceID)
 	}
 
+	marshaled, err := json.Marshal(inst)
+	unmarshaledInstance := internal.Instance{}
+	err = json.Unmarshal(marshaled, &unmarshaledInstance)
+
```
Comment (jaroslaw-pieszka) [inline components/kyma-environment-broker/internal/storage/driver/memory/instance.go:160]: What is the point?

Comment (ralikio) [reply components/kyma-environment-broker/internal/storage/driver/memory/instance.go:160]: Added a comment explaining the reasoning behind the code.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**